app/views/hostgroups/_form.html.erb

```diff
@@ -27,10 +27,19 @@
       <%= text_f f, :name %>
       <%= textarea_f f, :description, :rows => :auto %>
       <%= hostgroup_puppet_environment_field f %>
-      <%= select_f(f, :compute_profile_id, ComputeProfile.visibles, :id, :name,
+
+      <%= select_f f, :compute_resource_id, accessible_resource(@hostgroup, :compute_resource), :id, :to_label,
+          { :include_blank => blank_or_inherit_f(f, :compute_resource, :inherited_blank_value => _('Bare Metal'), :blank_value => _('Bare Metal')) },
+          { :label => _('Deploy on'),
+            :onchange => 'tfm.hostgroups.computeResourceChanged(this);',
+            :help_inline => :indicator } if SETTINGS[:unattended] %>
+
+      <div id="compute_profile" <%= display?(!@hostgroup.compute_resource_id) %> >
```

Comment:
I wonder if we should actually hide this if there is no compute resource set - maybe users want to just set the profile at the hostgroup level (as they can already)? 
Also, this isn't displayed if the compute resource is inherited from the parent.